Source/Core/VideoBackends/Vulkan/Renderer.cpp

```diff
@@ -561,6 +561,19 @@ void Renderer::SetScissorRect(const MathUtil::Rectangle<int>& rc)
 {
   VkRect2D scissor = {{rc.left, rc.top},
                       {static_cast<u32>(rc.GetWidth()), static_cast<u32>(rc.GetHeight())}};
+
+  // See Vulkan spec for vkCmdSetScissor:
+  // The x and y members of offset must be greater than or equal to 0.
+  if (scissor.offset.x < 0)
+  {
+    scissor.extent.width -= -scissor.offset.x;
```

Comment:
This seems a bit strange to me, doesn't this mean the scissor rect is actually smaller than initially requested? Should this be a translation along the negative valued axis instead?

Reply:
The parameter is a rectangle, so left/right/top/bottom, not offset/width. The Vulkan struct is initially populated with the width of the rectangle, without clamping to zero. Consider a rect of say, -10-20, that's going to be 0-20 on screen, but the width is 30. So by setting the offset to 0, we have to compensate by reducing the width by 10.

Reply:
Makes sense 👍